### ISSUE TITLE
Voiceover not reading elements for nested DialogFullScreen component in Safari browser

### ISSUE THREAD

**Issue by kishanivelupillai-sage** [Sage/carbon#5303]
### Current behaviour

When a child `DialogFullScreen` component is nested within a parent `DialogFullScreen` component, Voiceover does not read out elements in the child `DialogFullScreen` component using Safari browser.

### Expected behaviour

When a child `DialogFullScreen` component is nested within a parent `DialogFullScreen` component, expect Voiceover to read out elements in the child `DialogFullScreen` component using Safari browser.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/voiceover-dialog-i64x57?file=/src/index.js

### JIRA Ticket (Sage Only)

SBS-31787

### Carbon Version

109.0.0

### What browsers are you seeing the problem on?

Safari

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
